app/screens/forgot_password/forgot_password.js

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import Button from 'react-native-button';
+import {intlShape} from 'react-intl';
+
+import {
+    Image,
+    Keyboard,
+    StyleSheet,
+    Text,
+    TextInput,
+    TouchableWithoutFeedback,
+    View,
+} from 'react-native';
+
+import {GlobalStyles} from 'app/styles';
+
+import ErrorText from 'app/components/error_text';
+import FormattedText from 'app/components/formatted_text';
+import StatusBar from 'app/components/status_bar';
+
+import {isEmail} from 'app/utils/general';
+
+export default class ForgotPassword extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            sendPasswordResetEmail: PropTypes.func.isRequired,
+        }),
+    }
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            error: null,
+            email: '',
+            sentPasswordLink: false,
+        };
+    }
+
+    changeEmail = (email) => {
+        this.setState({
+            email,
+        });
+    }
+
+    submitResetPassword = async () => {
+        if (!this.state.email || !isEmail(this.state.email)) {
+            const {formatMessage} = this.context.intl;
+            this.setState({
+                error: formatMessage({id: 'password_send.error', defaultMessage: 'Please enter a valid email address.'}),
+            });
+            return;
+        }
+
+        const {data, error} = await this.props.actions.sendPasswordResetEmail(this.state.email);
+        if (error) {
+            this.setState({error});
+        } else if (this.state.error) {
+            this.setState({error: ''});
+        }
+        if (data) {
+            this.setState({sentPasswordLink: true});
+        }
+    }
+
+    emailIdRef = (ref) => {
+        this.emailId = ref;
+    };
+
+    blur = () => {
+        if (this.emailId) {
+            this.emailId.blur();
+        }
+        Keyboard.dismiss();
```

Comment:
I don't think this is necessary since the keyboard should be dismissed automatically when the input loses focus.